client/app/collections/filtered.coffee

```diff
@@ -4,12 +4,22 @@ ContactViewModel = require 'views/models/contact'
 
 tagRegExp = search.pattern 'tag'
 app       = null
-scores    = new WeakMap()
+scores    = new Map()
+
+
+Math.median = (array) ->
+    sorted = array.sort (a, b) -> a - b
+    len = array.length
+
+    if len % 2
+        sorted[Math.floor len / 2]
+    else
+        (sorted[len / 2] + sorted[len / 2 - 1]) / 2
```

Comment:
The median is not unique in that case. For example, `[1,2,3,4]` can have anything between 2 and 3 as median. In practice, taking 3 is easier than computing the mean of 2 and 3, and works as well.